### PR TITLE
Separate behavior for students and teachers for /users, permissions for patch /users/{id}

### LINKED ISSUE #23: Separate behavior for teachers and students for /users, add remaining fields to model
## For all endpoints
`/users` endpoints should behave differently for students and teachers. For students the responses should look like:
```json
{
  "status": "success",
  "data": {
    "user": {
      "id": 1,
      "firstName": "Alex",
      "lastName": "Walker",
      "userName": "Alex Walker",
      "githubUsername": "alex-walker",
      "email": "alex.walker@boolean.co.uk",
      "mobile": "07890 123456",
      "role": "STUDENT",
      "specialism": "Software Developer",
      "cohort_id": 4,
      "start_date": "mm-yyyy",
      "end_date": "mm-yyyy",
      "biography": "Hello there!",
    }
  }
}
```

For teachers:
```json
{
  "status": "success",
  "data": {
    "user": {
      "id": 2,
      "firstName": "Dylan",
      "lastName": "Green",
      "userName": "Dylan Green",
      "githubUsername": "dylan-green",
      "email": "teacher@test.com",
      "mobile": "07654 321098",
      "role": "TEACHER",
      "specialism": "Software Developer",
      "job_title": "Software Development Instructor",
      "biography": "Hello there!",
    }
  }
}
```

## Patch endpoint
For the patch endpoint only users logged in as teachers should be able to change the `cohort_id` or `role` of other users, students should only be able to change their profile credentials. Students should receive a `403` when attempting to update fields they're not permitted to

## Remaining model changes
To reflect the usage in the frontend the `githubUrl` field should be renamed to `githubUsername`. Also the fields `startDate` and `endDate` needs to be added for students.

### PR DESCRIPTION
**Pull request by gremble0**
https://github.com/orgs/boolean-uk/projects/19/views/2?pane=issue&itemId=85181387&issue=boolean-uk%7Cteam-dev-server-ex-2410-team1%7C23

### CODE REVIEW

**Review by thomaafl — APPROVED** [review]

**Review by johnfa1508 — APPROVED** [review]
Looks good Herman

File: src/controllers/user.js [hunk 1]
```diff
@@ -88,34 +90,42 @@ export const updateById = async (req, res) => {
...
-      'specialism',
-      'imageUrl',
-      'cohortId'
-    ]
-
-    allowedUpdates.forEach((field) => {
-      if (updates[field] !== undefined) {
-        existingUser[field] = updates[field]
+    const updateUnchecked = async () => {
+      if (updates.password) {
```
Comment (dogezen) [inline src/controllers/user.js:94]: are you going to validate that the password matches the length, character requirements?

Comment (dogezen) [reply src/controllers/user.js:94]: what happens if I submit a password "a" via an api call?

**Review by dogezen — APPROVED** [review]
Have left a comment about password validation, you might want to add validations server side for password, email, you can mark these as 1 new issue you work on separately from this

**Pull request merged**